### DISCUSSION TITLE
I can't see create 3 topics when using the university network

### DISCUSSION THREAD

**Discussion by nidia-bucarelli** [iRobotEducation/create3_docs#263] [Questions]
I acquired the Turtlebot4 (built based on the iRobot Create3) and I have issues with the university WIFI connection. I can connect the robot/Pi4, Create, and my computer to the university WIFI. The issue arises with it being able to full communicate. For example, I cannot see the robot topics from my computer. The robot uses multicast, which the university seems to be blocking. I tried to disable the multicast of cyclonedss with the file provided on the iRobot website. I added the file to both the Pi4 and my computer. My files are below ( I also tried to add the Create as a peer in both files, but it didn't work as well).
After disabling multicast, I can subscribe and publish between the Pi4 and my computer, but when I run ros2 topic list I cannot see the topics of the robot. I am not sure what I am doing wrong.

I would appreciate any help you can provide!

![imagen](https://user-images.githubusercontent.com/112603498/205205220-87214d09-1ef9-49d6-aa5a-45725643fa24.png)

![imagen](https://user-images.githubusercontent.com/112603498/205205246-b6de4963-02ce-44f9-acaf-faece2d8af41.png)

**Comment by brianabouchard** [comment]
Hello! I'm happy to help. I've tested quite a number of configurations with the Create 3 Robot on university wifi with similar restrictions to what it sounds like you are experiencing. But, I'm not quite as familiar with the Turtlebot 4, so please bear with me to make sure I understand the setup. When you connect with the Pi4, are you SSHing into it or are you 

In general, I've found that for the middleware (RMW) Fast-DDS works better when disabling multicasting, and my understanding is that unicast is better supported on this RMW. So, my first recommendation would be to switch to Fast. To do this, you'll have to enter `nano ~/.bashrc` in terminal (you'll have to do this on both your computer and then also your Pi4 via SSH) and then follow the steps outlined here: https://iroboteducation.github.io/create3_docs/setup/xml-config/ to modify the variables needed to use Fast-DDS. 

Then when you are filling in the XML file for Fast, I would include the IP addresses of your Create 3, Pi4 or Computer (depending on which you are loading the file onto - Pi4 IP for computer XML; computer IP for Pi4 XML), and local host IP. I've created a template for you below that allows you to add these three IP addresses to the standard XML file. It also looks like you are using the ethernet over USB option (usb0) for your Pi4 to Create 3 connection. In this case, the robot's IP that only the Pi can communicate with over usb0 would be 192.168.186.2. I've added this to the template below but it isn't needed on the computer because the computer won't be able to communicate with that IP. 

```
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
   <participant profile_name="unicast_connection" is_default_profile="true">
       <rtps>
           <builtin>
               <metatrafficUnicastLocatorList>
                   <locator/>
               </metatrafficUnicastLocatorList>
               <initialPeersList>
                   <locator>
                       <udpv4>
                           <address>ROBOT_IP</address>
                       </udpv4>
                   </locator>
                   <locator>
                       <udpv4>
                           <address>PI4_IP/COMPUTER_IP</address>
                       </udpv4>
                   </locator>
                   <locator>
                       <udpv4>
                           <address>LOCALHOST_IP</address>
                       </udpv4>
                   </locator>
                   <locator>
                       <udpv4>
                           <address>192.168.186.2</address>
                       </udpv4>
                   </locator>
               </initialPeersList>
           </builtin>
       </rtps>
   </participant>
</profiles>
```

That should work, but if it doesn't, let me know and we'll work through isolating each component to get a better idea of where the connection is failing!
